### PR TITLE
Keep Cassandra storage working even if the annotations query is dropped

### PR DESCRIPTION
**Pull request by michaelsembwever**
It should be possible to drop just the CONTAINS SASI and have Zipkin continue to start up.

(Trying to search by annotation will still ofcourse throw an exception, here a NPE)

### CODE REVIEW

**Comment by eirslett** [comment]
Maybe we can ship the NPE as a Server 500 error with a substatus that indicates it's due to SASI being dropped? That could eventually be used in the UI to show a sane error message.

File: zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/CassandraSpanStore.java [hunk 1]
```diff
@@ -73,6 +76,15 @@
     }
   }
 
+  private static SelectTraceIdsFromSpan.Factory initialiseSelectTraceIdsFromSpan(Session session) {
```
Comment (codefromthecrypt) [inline zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/CassandraSpanStore.java]: maybe add a comment here with the same rationale as the PR desc (which this allows dropping the SASI fields)

**Comment by michaelsembwever** [comment]
> Maybe we can ship the NPE as a Server 500 error with a substatus that indicates it's due to SASI being dropped? That could eventually be used in the UI to show a sane error message.

Good idea. A IllegalStateException with an appropriate message is now thrown.

File: zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/CassandraSpanStore.java (hunk 1 above)
Comment (michaelsembwever) [reply zipkin-storage/zipkin2_cassandra/src/main/java/zipkin2/storage/cassandra/CassandraSpanStore.java]: done.

**Comment by codefromthecrypt** [comment]
thx!

**Pull request merged**